pyinfra/connectors/util.py

```diff
@@ -13,7 +13,7 @@
 
 SUDO_ASKPASS_ENV_VAR = "PYINFRA_SUDO_PASSWORD"
 SUDO_ASKPASS_COMMAND = r"""
-temp=$(mktemp /tmp/pyinfra-sudo-askpass-XXXXXXXXXXXX)
+temp=$(mktemp "${{TMPDIR:=/tmp}}/pyinfra-sudo-askpass-XXXXXXXXXXXX")
```

Comment:
I have a feeling python formatting might make this line behave weirdly although I'm not quite sure yet I will test to confirm when I get the chance.

Reply:
The [double braces are how you escape braces inside format strings](https://peps.python.org/pep-3101/#format-strings). I've tested this code. So it ends up as:

```sh
temp=$(mktemp "${TMPDIR:=/tmp}/pyinfra-sudo-askpass-XXXXXXXXXXXX")
```

Which is the shell syntax that we want. It says "Use the value of the TMPDIR environment variable if set, otherwise use `/tmp`, followed by `/pyinfra-sudo-askpass-XXXXXXXXXXXX`" which is all passed to `mktemp` as a single argument, then assign the stdout of the `mktemp` command to the `temp` variable.

Reply:
Aha, I have never known that! Thank you for the explanation 👍